### PR TITLE
Adds the Fundamentally Evil quirk. Interactions with Empathy and Honorbound.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Adds the Fundamentally Evil quirk. The quirk does nothing in of itself.

If an Empath examines you, they will be shaken up by the fact that you are totally evil. 

Mindreaders can literally see that you're evil. If they're ALSO evil, it gives a gives a unique message.

Honorbound chaplains can freely attack you without first declaring you evil. You are already evil. There is no outward tells for this fact, but if the chaplain pays attention to the medical records, you might have a problem on your hands.

You are significantly harmed by holy water and holy explosions. So don't drink holy water if you can help it.

## Why It's Good For The Game

I just think it'd be funny to have some crew, for whatever reason, be fully committed to being evil for no reason other than it is funny to bother empaths.

I want a chaplain to come to the realization that I'm evil and see what happens.

## Changelog
:cl:
add: Fundamentally Evil quirk. You might act normal, but you know deep down that you totally don't give a shit about anyone but yourself. Empaths better watch out.
/:cl:

### CODE REVIEW

**Comment by DrAmazing343** [comment]
In exchange, make Empath help me make moods better. Somehow. Please. Even if examining some Fundamentally Evil motherfucker immediately nukes my real-life house from orbit, I want Empath to be at all worth taking ;w;

**Comment by carlarctg** [comment]
add mind reader interaction

File: code/datums/quirks/neutral_quirks/evil.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	desc = "Where you would have a soul is but an ink-black void. While you are committed to maintaing your social standing, \
+		anyone who stares too long into your cold, uncaring eyes will know the truth. You are truly evil. There is nothing \
+		wrong with you. You chose to be evil, committed to it. Your ambitions come first above all."
+	icon = FA_ICON_HAND_MIDDLE_FINGER
+	value = 0
+	mob_trait = TRAIT_EVIL
+	gain_text = span_notice("You shed what little remains of your humanity. You have work to do.")
+	lose_text = span_notice("You suddenly care more about others and their needs.")
+	medical_record_text = "Patient has passed all our social fitness tests with flying colours, but had trouble on the empathy tests."
+	mail_goodies = list(/obj/item/food/grown/citrus/lemon)
```
Comment (carlarctg) [inline code/datums/quirks/neutral_quirks/evil.dm:12]: ?

Comment (necromanceranne) [reply code/datums/quirks/neutral_quirks/evil.dm:12]: **Every** **V**illain **I**s **L**emons

**Comment by Loafin34** [comment]
can mail goodies also be the big red button?

File: code/datums/quirks/neutral_quirks/evil.dm (hunk 1 above)
Comment (Y0SH1M4S73R) [reply code/datums/quirks/neutral_quirks/evil.dm:12]: Otherwise known as **EVIL**

**Comment by OuroborusReporter** [comment]
Probably a bit too much, but given that this is a Dark And Twisted Mind quirk, having it set your Telepathy text to red like emagged megaphones would be neat

**Comment by necromanceranne** [comment]
I reduced the time empaths are negatively affected by seeing an evil dude, and also I made fundamentally evil guys die to holy water because that's funny.

**Comment by Loafin34** [comment]
can cultists with red eyes be given the fundamentally evil quirk? that'd be pretty funny

File: code/datums/quirks/neutral_quirks/evil.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/quirks/neutral_quirks/evil.dm:12]: wtf

**Comment by zoomachina** [comment]
> can cultists with red eyes be given the fundamentally evil quirk? that'd be pretty funny

no cultists are fundamentally good

**Comment by OceanFish1** [comment]
kaz i'm already a demon

**Comment by necromanceranne** [comment]
> can cultists with red eyes be given the fundamentally evil quirk? that'd be pretty funny

No.

**Comment by IndieanaJones** [comment]
Now that holy water harms people who have this perk, this is going to play into silicon politics regarding cult. If there's a human with this quirk who's also a cultist, you can't let them be deconverted under the pretense of human harm.

**Comment by necromanceranne** [comment]
> Now that holy water harms people who have this perk, this is going to play into silicon politics regarding cult. If there's a human with this quirk who's also a cultist, you can't let them be deconverted under the pretense of human harm.

Sure.

**Comment by necromanceranne** [comment]
Take that up with the admins.

**Comment by Everturning** [comment]
selfantag trait real

**Comment by DaCoolBoss** [comment]
Any way we can tie this to the "Champion of Evil" traitor faction?

**Comment by necromanceranne** [comment]
There is no way in hell I'm letting the trait get added by an antagonist role. If you wouldn't have the quirk in the medical record, it means you're an antag.

**Comment by necromanceranne** [comment]
> Hey got a question, not an ideaguy: is this a nuetral, positive, or negative quirk?

It is neutral.

**Comment by CCC23** [comment]
https://youtu.be/rhbZfG3Tq7M?si=37QPNBcD_jdD1-ld

**Comment by Loafin34** [comment]
> There is no way in hell I'm letting the trait get added by an antagonist role. If you wouldn't have the quirk in the medical record, it means you're an antag.

i had it specifically on red eyes because that's the moment where the cult can actually be spotted and an empath recoiling from fear of a mysteriously masked man sounds fun

**Comment by necromanceranne** [comment]
> i had it specifically on red eyes because that's the moment where the cult can actually be spotted and an empath recoiling from fear of a mysteriously masked man sounds fun

I am not going to make Empath a valid hunting quirk against cultists. Neither should anyone else.

**Comment by Everturning** [comment]
halo, however....

**Comment by necromanceranne** [comment]
No.

**Comment by MrMelbert** [comment]
I think this is going to be the most bikesheded PR created

**Comment by necromanceranne** [comment]
You give me too much credit.

**Comment by moocowswag** [comment]
Does it persist via mind so say a changeling disguises as someone else and now that person is suddenly taking damage from holy water or able to be hit by chap or getting a special message when mindread?

**Comment by DaydreamIQ** [comment]
Considering this is a neutral quirk there's no downside to taking this if your goal mess with security and scream "AI HELP SEC IS POISONING ME" when they try to deconvert you during cult, on top of it making deconversion kind of a pain in the ass since dead people don't process chems. Like you can say to just let the admins sort that out but that's a really poor mindset to have in general

**Comment by Namwty** [comment]
> Considering this is a neutral quirk there's no downside to taking this if your goal mess with security and scream "AI HELP SEC IS POISONING ME" when they try to deconvert you during cult, on top of it making deconversion kind of a pain in the ass since dead people don't process chems. Like you can say to just let the admins sort that out but that's a really poor mindset to have in general

Self harm rule etc etc, when it comes to silicons

if you take a quirk that makes you die to *thing*, it's your choice. A.K.A. don't get mad at admins/players when taking a quirk that paints a straight up target on your back. Like along with blind quirk etc, no one is going to babysit you with it, it's a **choice** to take the quirk. TL;DR - IC issue, + maybe note for ban/note baiting.


Regardless, we getting out of Ma'habre with this quirk, for sure.

**Comment by DaydreamIQ** [comment]
Saying its self harm is the highest degree of cope imaginable. That's like saying force-feeding someone medicine or food someone's allergic to isn't harmful. Holy water is POISON to anyone taking this quirk and they quite literally won't be ingesting it themselves

**Comment by Jackal-boop** [comment]
May wanna give this a "this isn't a lisence to grief" text like RDS has

**Comment by necromanceranne** [comment]
> Does it persist via mind so say a changeling disguises as someone else and now that person is suddenly taking damage from holy water or able to be hit by chap or giving a special message when mindread?

Are you...talking about a changeling making someone _else_ register as evil? Or whether or not the ling would still react the same way when examined? 

> Considering this is a neutral quirk there's no downside to taking this if your goal mess with security and scream "AI HELP SEC IS POISONING ME" when they try to deconvert you during cult, on top of it making deconversion kind of a pain in the ass since dead people don't process chems. Like you can say to just let the admins sort that out but that's a really poor mindset to have in general

If they can handle situations where players die due to allergies, I'm sure they can handle the holy water allergy.

**Comment by necromanceranne** [comment]
> May wanna give this a "this isn't a lisence to grief" text like RDS has

Investigating the reason that immersion breaking popup even happens, it's because the author of quirks all that time ago was being overcautious about players somehow interpreting 'insane = kill everyone' and had a deep mistrust of players/admins handling it properly. Until the exact moment someone it is a problem, I won't worry about it.

**Comment by carlarctg** [comment]
i think if someone is dumb enough to think that evil quirk means grief license they should be allowed to have that misunderstanding happen, so they can have a note for being a baboon

**Comment by carlarctg** [comment]
Also, I don't think holy water should hurt. Not because of bikeshedding policy and metagaming but because chaplains are not necessarily (or even commonly) good. A morbid, scythe wielding chaplain that prays to the god of greed being hurt by his own blessed water, because he's evil? Odd.

File: code/datums/quirks/neutral_quirks/evil.dm (hunk 1 above)
Comment (theselfish) [reply code/datums/quirks/neutral_quirks/evil.dm:12]: image

**Comment by necromanceranne** [comment]
Listen, even the most evil chaplains make 'holy water', it's not really something I'm looking to shake up at the moment.

**Comment by Xkallubar** [comment]
> Also, I don't think holy water should hurt. Not because of bikeshedding policy and metagaming but because chaplains are not necessarily (or even commonly) good. A morbid, scythe wielding chaplain that prays to the god of greed being hurt by his own blessed water, because he's evil? Odd.

It's also kind of forces the notion of "Holy water = good, therefore, chaplain = good"
What if I wanted to be an evil chaplain? Why would an evil chaplain see something that literally kills them as holy?

**Comment by Xkallubar** [comment]
Please make chaplains that take this trait immune to the downsides.

**Comment by MercuryArrow** [comment]
> It's also kind of forces the notion of "Holy water = good, therefore, chaplain = good" What if I wanted to be an evil chaplain? Why would an evil chaplain see something that literally kills them as holy?

I mean, evil chaplains already take damage from unholy water and eldritch essence, even though we have an alignment system built into the sects that specifically flags some as evil and some as good.  It's always been a little inconsistent and rough around the edges and dealing with that whole mess is probably out of scope for now.

**Comment by zoomachina** [comment]
> I mean, evil chaplains already take damage from unholy water and eldritch essence, even though we have an alignment system built into the sects that specifically flags some as evil and some as good. It's always been a little inconsistent and rough around the edges and dealing with that whole mess is probably out of scope for now.

unholy water and eldritch essence are antag healing items, not potions that heal everyone who's evil, while holy water is chaplain item, not 'good potion', and the notion that this has to be changed at some point sounds completely alien to me

**Comment by DrAmazing343** [comment]
god's strongest debate on a small fun PR. I think it's good as-is, unless someone adds a quirk to let the Chap get fucking Unholy Water somehow or something silly like that then I think Fundamentally Evil people SHOULD get hurt even if they are "holy."

File: code/modules/mob/living/carbon/examine.dm [hunk 2]
```diff
@@ -219,6 +219,10 @@
 					. += "[t_He] [t_is] flushed and wheezing."
 				if (bodytemperature < dna.species.bodytemp_cold_damage_limit)
 					. += "[t_He] [t_is] shivering."
+				if(HAS_TRAIT(src, TRAIT_EVIL))
+					. += "[t_His] eyes radiate with a unfeeling, cold detachment. There is nothing but darkness within [t_His] soul."
```
Comment (Time-Green) [inline code/modules/mob/living/carbon/examine.dm:223]:
```suggestion
					. += "[t_His] eyes radiate with a unfeeling, cold detachment. There is nothing but darkness within [t_his] soul."
```

File: code/datums/mutations/antenna.dm [hunk 3]
```diff
@@ -96,6 +96,11 @@
 		to_chat(owner, span_warning("You plunge into your mind... Yep, it's your mind."))
 		return
 
+	if(HAS_TRAIT(cast_on, TRAIT_EVIL))
+		to_chat(owner, span_warning("As you reach into [cast_on]'s mind, \
+			you feel the overwhelming emptiness within. A truly evil being. \
+			[HAS_TRAIT(owner, TRAIT_EVIL) ? "It's nice to find someone of like-mind." : "What is wrong with this person?"]"))
```
Comment (Time-Green) [inline code/datums/mutations/antenna.dm:102]:
```suggestion
			[HAS_TRAIT(owner, TRAIT_EVIL) ? "It's nice to find someone who is like-minded." : "What is wrong with this person?"]"))
```

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Review by Time-Green — APPROVED** [review]

File: code/datums/quirks/neutral_quirks/evil.dm [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/datum/quirk/evil
+	name = "Fundamentally Evil"
+	desc = "Where you would have a soul is but an ink-black void. While you are committed to maintain your social standing, \
```
Comment (ATH1909) [inline code/datums/quirks/neutral_quirks/evil.dm:3]:
```suggestion
	desc = "Where you would have a soul is but an ink-black void. While you are committed to maintaining your social standing, \
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -411,6 +412,9 @@
 				affected_mob.Unconscious(12 SECONDS)
 				to_chat(affected_mob, span_cult_large("[pick("Your blood is your bond - you are nothing without it", "Do not forget your place", \
 					"All that power, and you still fail?", "If you cannot scour this poison, I shall scour your meager life!")]."))
+		else if(HAS_TRAIT(affected_mob, TRAIT_EVIL) && SPT_PROB(25, seconds_per_tick)) //Congratulations, your committment to evil has now made holy water a deadly poison to you!
+			affected_mob.emote("scream")
+			need_mob_update += affected_mob.adjustBruteLoss(3 * REM * seconds_per_tick, updating_health = FALSE)
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:417]: should be burn damage, no?

File: code/modules/religion/honorbound/honorbound_trauma.dm [hunk 6]
```diff
@@ -73,6 +73,8 @@
 		guilty(attacked_mob, "for blasphemous magicks!")
 	if(HAS_TRAIT(attacked_mob, TRAIT_CULT_HALO))
 		guilty(attacked_mob, "for blasphemous worship!")
+	if(HAS_TRAIT(attacked_mob, TRAIT_EVIL))
+		guilty(attacked_mob, "an almost fanatical committment to EEEEVIL!")
```
Comment (ATH1909) [inline code/modules/religion/honorbound/honorbound_trauma.dm:77]:
```suggestion
		guilty(attacked_mob, "an almost fanatical commitment to EEEEVIL!")
```

File: code/datums/quirks/neutral_quirks/evil.dm (hunk 1 above)
Comment (ATH1909) [reply code/datums/quirks/neutral_quirks/evil.dm:12]: amazing

**Comment by ATH1909** [comment]
> Now that holy water harms people who have this perk, this is going to play into silicon politics regarding cult. If there's a human with this quirk who's also a cultist, you can't let them be deconverted under the pretense of human harm.

you could fix (at least in the red eyes/halo stage) this by making holy water not damage evil people while they're cultists and making the deconversion process instantly purge the holy water in your body when it completes.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 5 above)
Comment (ATH1909) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:417]: same with the other place they take brute damage lower down

Comment (necromanceranne) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:417]: oh you're right, it should be

**Comment by necromanceranne** [comment]
> > Now that holy water harms people who have this perk, this is going to play into silicon politics regarding cult. If there's a human with this quirk who's also a cultist, you can't let them be deconverted under the pretense of human harm.
> 
> you could fix this (at least in the red eyes/halo stage) by making holy water not damage evil people while they're cultists and making the deconversion process instantly purge the holy water in your body when it completes.

Thankfully, holy water already self purges once you've hit the point of deconversion. But yes, I've changed it so that if you're a cultist or chaplain you won't be hurt even if you have the evil trait.

**Comment by Dmeto** [comment]
If this was the counter to friendly, I would suggest making the indivuals heart smaller and replace love with hauntium. Tho I might PR this in post.

**Comment by necromanceranne** [comment]
sounds kinda cool

**Comment by JoshAdamPowell** [comment]
This is very cool!

**Comment by DaCoolBoss** [comment]
> What if evil people could identify "weak" people, like if they had social anxiety, pushover, frail or whatever

Not gonna happen, detecting traits is a medbay/psychologist thing.

> There is no way in hell I'm letting the trait get added by an antagonist role. If you wouldn't have the quirk in the medical record, it means you're an antag.

I didn't mean that, I meant the opposite. Evil trait increasing the odds of rolling "champion of evil" when the traitor antag looks for a faction to assign. Possibly reducing the odds or stopping you rolling legal trouble or animal/robot rights terrorist? There's a few traits now that could conceivably impact your "backstory", and I think it might go a ways to making players feel like their assigned faction is relevant to their character rather than just arbitrary. Might be a job for another PR?

**Comment by necromanceranne** [comment]
Didn't Champion of Evil get removed recently?

**Pull request merged**